internal/ethapi/transaction_args.go

```diff
@@ -108,7 +108,7 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 				return err
 			}
 			if b.ChainConfig().IsLondon(head.Number) {
-				price.Add(price, head.BaseFee)
+				price.Add(price, new(big.Int).Mul(head.BaseFee, big.NewInt(2)))
```

Comment:
Could you elaborate why we add basefee multiplied by `2` to the price here?

Reply:
That what the above branch defaults `MaxFeePerGas` to, so I assume he was just following that patter?  https://github.com/ethereum/go-ethereum/blob/8fe47b0a0d4280020c246fee817a87e358f868f3/internal/ethapi/transaction_args.go#L94

Reply:
Yes. The idea is if the `MaxFeePerGas` is not set after the london, we add the 2*baseFee to the user gas price by default.

Reply:
I think the legacy tx gas price suggestion should not add 2x base fee because all fees are consumed, so it would result in a spiral upwards.

Reply:
Fixed